src/axom/primal/tests/primal_integral.cpp

```diff
@@ -0,0 +1,231 @@
+  primal::CurvedPolygon<double, 2> parabola_polygon(pedges, 2);
+  EXPECT_NEAR(evaluate_area_integral(parabola_polygon, const_integrand, npts),
+              8.0 / 3.0,
+              abs_tol);
+  EXPECT_NEAR(evaluate_area_integral(parabola_polygon, poly_integrand, npts),
+              64.0 / 105.0,
+              abs_tol);
+  EXPECT_NEAR(evaluate_area_integral(parabola_polygon, transc_integrand, npts),
+              0.0,
+              abs_tol);
+}
+
+TEST(primal_integral, evaluate_line_integral_scalar)
+{
+  using Point2D = primal::Point<double, 2>;
+  using Bezier = primal::BezierCurve<double, 2>;
+  double abs_tol = 1e-10;
+
+  // Quadrature nodes. Should be sufficiently high to pass tests
+  int npts = 30;
+
+  // Define anonymous functions for testing
+  auto const_integrand = [](Point2D /*x*/) -> double { return 1.0; };
+  auto poly_integrand = [](Point2D x) -> double { return x[0] * x[1] * x[1]; };
+  auto transc_integrand = [](Point2D x) -> double {
+    return std::sin(x[0] * x[1]);
+  };
+
+  // Test on single parabolic segment
+  Point2D paranodes[] = {Point2D {-1.0, 1.0},
+                         Point2D {0.5, -2.0},
+                         Point2D {2.0, 4.0}};
+  Bezier parabola_segment(paranodes, 2);
+
+  // Compare against hand computed/high-precision calculated values.
+
+  // Constant integrand line integral is equivalent to arc-length calculation
+  EXPECT_NEAR(evaluate_line_integral(parabola_segment, const_integrand, npts),
+              6.12572661998,
+              abs_tol);
+
+  EXPECT_NEAR(evaluate_line_integral(parabola_segment, poly_integrand, npts),
+              37.8010703669,
+              abs_tol);
+  EXPECT_NEAR(evaluate_line_integral(parabola_segment, transc_integrand, npts),
+              0.495907795678,
+              abs_tol);
+
+  // Test on a collection of Bezier curves
+  Point2D segnodes1[] = {Point2D {-1.0, -1.0},
+                         Point2D {-1.0 / 3.0, 1.0},
+                         Point2D {1.0 / 3.0, -1.0},
+                         Point2D {1.0, 1.0}};
+  Bezier cubic_segment(segnodes1, 3);
+
+  Point2D segnodes2[] = {Point2D {1.0, 1.0}, Point2D {-1.0, 0.0}};
+  Bezier linear_segment(segnodes2, 1);
+
+  Point2D segnodes3[] = {Point2D {-1.0, 0.0},
+                         Point2D {-3.0, 1.0},
+                         Point2D {-1.0, 2.0}};
+  Bezier quadratic_segment(segnodes3, 2);
+
+  axom::Array<Bezier> connected_curve(
+    {cubic_segment, linear_segment, quadratic_segment});
+
+  // Compare against hand computed/high-precision calculated values
+  EXPECT_NEAR(evaluate_line_integral(connected_curve, const_integrand, npts),
+              8.28968500196,
+              abs_tol);
+  EXPECT_NEAR(evaluate_line_integral(connected_curve, poly_integrand, npts),
+              -5.97565740064,
+              abs_tol);
+  EXPECT_NEAR(evaluate_line_integral(connected_curve, transc_integrand, npts),
+              -0.574992518405,
+              abs_tol);
+}
```

Comment:
Please add a test with disconnected line segments, as this is noted as possible in the doxygen for `evaluate_line_integral`.  Perhaps displace the middle, linear segment by (0, 2) or some such.